delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViews.scala

```diff
@@ -1,17 +1,313 @@
+    } yield res
+  }.named("updateView", moduleType)
+
+  /**
+    * Add a tag to an existing view.
+    *
+    * @param id       the id of the view
+    * @param project  the project to which the view belongs
+    * @param tag      the tag label
+    * @param tagRev   the target revision of the tag
+    * @param rev      the current revision of the view
+    */
+  def tag(
+      id: IdSegment,
+      project: ProjectRef,
+      tag: TagLabel,
+      tagRev: Long,
+      rev: Long
+  )(implicit subject: Subject): IO[BlazegraphViewRejection, BlazegraphViewResource] = {
+    for {
+      p   <- projects.fetchActiveProject(project)
+      iri <- expandIri(id, p)
+      res <- eval(TagBlazegraphView(iri, project, tagRev, tag, rev, subject), p)
+    } yield res
+  }.named("tagView", moduleType)
+
+  /**
+    * Deprecate a view.
+    *
+    * @param id       the view id
+    * @param project  the project to which the view belongs
+    * @param rev      the current revision of the view
+    */
+  def deprecate(
+      id: IdSegment,
+      project: ProjectRef,
+      rev: Long
+  )(implicit subject: Subject): IO[BlazegraphViewRejection, BlazegraphViewResource] = {
+    for {
+      p   <- projects.fetchActiveProject(project)
+      iri <- expandIri(id, p)
+      res <- eval(DeprecateBlazegraphView(iri, project, rev, subject), p)
+    } yield res
+  }.named("deprecateView", moduleType)
+
+  /**
+    * Fetch the latest revision of a view.
+    *
+    * @param id       the view id
+    * @param project  the project to which the view belongs
+    */
+  def fetch(
+      id: IdSegment,
+      project: ProjectRef
+  ): IO[BlazegraphViewRejection, BlazegraphViewResource] = fetch(id, project, None).named("fetchView", moduleType)
+
+  /**
+    * Fetch the view at a specific revision.
+    *
+    * @param id       the view id
+    * @param project  the project to which the view belongs
+    * @param rev      the revision to fetch
+    */
+  def fetchAt(
+      id: IdSegment,
+      project: ProjectRef,
+      rev: Long
+  ): IO[BlazegraphViewRejection, BlazegraphViewResource] =
+    fetch(id, project, Some(rev)).named("fetchViewAt", moduleType)
+
+  /**
+    * Fetch view by tag.
+    *
+    * @param id       the view id
+    * @param project  the project to which the view belongs
+    * @param tag      the tag to fetch
+    */
+  def fetchBy(id: IdSegment, project: ProjectRef, tag: TagLabel): IO[BlazegraphViewRejection, BlazegraphViewResource] =
+    fetch(id, project, None)
+      .flatMap { resource =>
+        resource.value.tags.get(tag) match {
+          case Some(rev) => fetchAt(id, project, rev).leftMap(_ => TagNotFound(tag))
+          case None      => IO.raiseError(TagNotFound(tag))
+        }
+      }
+      .named("fetchViewBy", moduleType)
+
+  /**
+    * List views.
+    *
+    * @param pagination the pagination settings
+    * @param params     filtering parameters for the listing
+    */
+  def list(
```

Comment:
I don't think we need this method. Listings from the API are going to be provided by elasticsearch plugin.

However it might be useful for other purposes, outside of the API.